aws_embedded_metrics/logger/metrics_context.py

```diff
@@ -79,6 +79,12 @@ def put_dimensions(self, dimensions: Dict[str, str]) -> None:
 
         self.validate_dimension_set(dimensions)
 
+        # Duplicate dimension sets are removed before being added to the end of the collection.
+        # This ensures only latest dimension value is used as a target member on the root EMF node.
+        # This operation is O(n^2), but acceptable given sets are capped at 30 dimensions
+        incoming_keys: Set = set(dimensions.keys())
+        self.dimensions = list(filter(lambda dim: (set(dim.keys()) != incoming_keys), self.dimensions))
```

Comment:
The following test doesn't pass:

```python
    context = MetricsContext()
    dimension1 = {'key1': 'val1'}
    dimension2 = {'key2': 'val2'}

    context.set_dimensions(dimension1)
    context.put_dimensions(dimension2)

    assert len(context.get_dimensions()) == 2
```

Reply:
Solved. Should be `context.set_dimensions([dimension1])`.